### PR TITLE
[CI]Add model basic correctness test(Qwen2.5_7B)

### PR DESCRIPTION
**Pull request by Potabk**
### What this PR does / why we need it?

This pull request  tests the basic correctness of Qwen2.5_7B by comparing the inference results under long sequences of vllm and hf, We fixed the long sequence accuracy issue at [#377](https://github.com/vllm-project/vllm-ascend/pull/377), so before this pr is merged , plz have [#377](https://github.com/vllm-project/vllm-ascend/pull/377) merged  into main

### CODE REVIEW

**Comment by Potabk** [comment]
and there has another question is: when passing dtype='auto', it works well and result can equal with hf result, but when `dtype='half'` passed in, the result not as expected, mentioned at [#374](https://github.com/vllm-project/vllm-ascend/issues/374)

File: tests/test_basic_correctness.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from conftest import VllmRunner
+from model_utils import check_outputs_equal
+
+MODELS = ["Qwen/Qwen2.5-7B-Instruct"]
+
+TARGET_TEST_SUITE = os.environ.get("TARGET_TEST_SUITE", "L4")
+
+
+@pytest.mark.parametrize("model", MODELS)
+@pytest.mark.parametrize("backend", ["ASCEND"])
```
Comment (MengqingCao) [inline tests/test_basic_correctness.py]: The out-of-tree platforms of vLLM do not have a `backend`, Iet's just remove this arg.

File: tests/conftest.py [hunk 2]
```diff
@@ -52,14 +63,14 @@ def __init__(
         task: TaskOption = "auto",
         tokenizer_name: Optional[str] = None,
         tokenizer_mode: str = "auto",
-        # Use smaller max model length, otherwise bigger model cannot run due
-        # to kv cache size limit.
```
Comment (MengqingCao) [inline tests/conftest.py:56]: let's keep the comments for `max_model_len`

File: tests/conftest.py [hunk 3]
```diff
@@ -329,3 +340,444 @@ def __exit__(self, exc_type, exc_value, traceback):
 @pytest.fixture(scope="session")
 def vllm_runner():
     return VllmRunner
+
+
+_T = TypeVar("_T", nn.Module, torch.Tensor, BatchEncoding, BatchFeature, dict)
+
+
+# Inspired from https://github.com/vllm-project/vllm/blob/main/tests/conftest.py
```
Comment (MengqingCao) [inline tests/conftest.py]: This have already been commented in license header
```suggestion
```

File: tests/test_basic_correctness.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+Run `pytest tests/basic_correctness/test_basic_correctness.py`.
+"""
+import os
+
+import pytest
+from conftest import VllmRunner
+from model_utils import check_outputs_equal
+
+MODELS = ["Qwen/Qwen2.5-7B-Instruct"]
+
```
Comment (MengqingCao) [inline tests/test_basic_correctness.py:30]: Let's use modelscope for better model download and keep the configuration of `PYTORCH_NPU_ALLOC_CONF` to ensure oom not raised in CI
```suggestion

os.environ["VLLM_USE_MODELSCOPE"] = "True"
os.environ["PYTORCH_NPU_ALLOC_CONF"] = "max_split_size_mb:256"
```

Comment (Potabk) [reply tests/test_basic_correctness.py:30]: I don't know if hfrunner have the same way to speed up model download, otherwise we will download two weights from two different repositories

File: tests/test_basic_correctness.py (hunk 1 above)
Comment (Potabk) [reply tests/test_basic_correctness.py]: fixed at [e5af81](https://github.com/vllm-project/vllm-ascend/commit/e5af8173d7b0db3a0c896174d85514b0ac6b490a)

File: tests/conftest.py (hunk 2 above)
Comment (Potabk) [reply tests/conftest.py:56]: fixed at [e5af81](https://github.com/vllm-project/vllm-ascend/commit/e5af8173d7b0db3a0c896174d85514b0ac6b490a)

File: tests/conftest.py (hunk 3 above)
Comment (Potabk) [reply tests/conftest.py]: fixed at [e5af81](https://github.com/vllm-project/vllm-ascend/commit/e5af8173d7b0db3a0c896174d85514b0ac6b490a)

File: tests/test_basic_correctness.py (hunk 4 above)
Comment (Potabk) [reply tests/test_basic_correctness.py:30]: fixed at [e5af81](https://github.com/vllm-project/vllm-ascend/commit/e5af8173d7b0db3a0c896174d85514b0ac6b490a)

Comment (MengqingCao) [reply tests/test_basic_correctness.py:30]: Make sense. We currently disable modelscope then, but if requesting hf to download the model will break CI, we need to fallback to this. hfrunner can use splicing local paths to prevent duplicate downloads.
